docs/authoring.md

```diff
@@ -0,0 +1,147 @@
+# Authoring Components
+
+## Overview
+C#/WinRT provides support for authoring Windows Runtime components. You can write a library in C#, and use C#/WinRT's source generator to get a winmd that any WinRT compatible language can use. For example, a library written in C# can be used by a C++ program, via C#/WinRT and C++/WinRT, with just a few tweaks to the C++ project.
+
+
+## References
+Here are some resources that demonstrate authoring C#/WinRT components and the details discussed in this document.
+1. https://github.com/microsoft/CsWinRT/tree/master/Authoring/AuthoringConsumptionTest
+2. https://github.com/AdamBraden/MyRandom
+
+
+## Authoring the C# Component
+To create a library, select the Class Library (.NET Core) template in Visual Studio. C#/WinRT projects require SDK specific .NET frameworks.
+
+Accepted `<TargetFramework>` properties |
+--- |
+`net5.0-windows10.0.19041.0` |
+`net5.0-windows10.0.18362.0` |
+`net5.0-windows10.0.17763.0` |
+
+The library you are authoring should specify the following properties in its project file: 
+```
+  <PropertyGroup>
+    <!-- update the Windows SDK version to reflect your TargetFramework -->
+    <CsWinRTWindowsMetadata>10.0.19041.0</CsWinRTWindowsMetadata>
+    <CsWinRTComponent>true</CsWinRTComponent>
+    <CsWinRTEnableLogging>true</CsWinRTEnableLogging>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <GeneratedFilesDir Condition="'$(GeneratedFilesDir)'==''">$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)', '$(IntermediateOutputPath)', 'Generated Files'))</GeneratedFilesDir>
+  </PropertyGroup>
+```
+And don't forget to include a `PackageReference` to `Microsoft.Windows.CsWinRT`!
+
+
+## Known Authoring Issues
+1. There are some programs you could write as/in your component that aren't available in the Windows Runtime. 
+We are working on implementing diagnostics in our tool that will catch these errors before a winmd is generated for your component.
+
+2. Not all C# types have been mapped to runtime types, but we are working on completing this coverage. 

```

Comment:
runtime -> WinRT.  Not all mappable native C# types haves have been mapped to WinRT types,...